### DISCUSSION TITLE
contextHolder

### DISCUSSION THREAD

**Discussion by hamedyosefian** [ant-design/ant-design#39528] [Q&A]
How to use contextHolder in app.js nextjs???
when use contextHolder show alert just in current page

```
  const [messageApi, contextHolder] = message.useMessage();

{contextHolder}

```

**Comment by MadCcc** [comment]
Just put it in your JSX.

**Comment by zombieJ** [comment]
https://ant.design/components/message#components-message-demo-hooks

**Comment by vitalijalbu** [comment]
Is there a way guys to put it globally in app? I use nextjs

**Reply by JulienSol** [reply]
Yes with https://ant.design/components/app
